axelrod/strategies/gambler.py

```diff
@@ -1,75 +1,100 @@
-from axelrod import Actions, Player, init_args, random_choice
-from axelrod.strategy_transformers import FinalTransformer
-from .lookerup import LookerUp, create_lookup_table_keys
+"""Stochastic variants of Lookup table based-strategies, trained with particle
+swarm algorithms.
+
+For the original see:
+ https://gist.github.com/GDKO/60c3d0fd423598f3c4e4
+"""
+
+from axelrod import Actions, random_choice, load_pso_tables
+from .lookerup import LookerUp, create_lookup_table_from_pattern
 
 
 C, D = Actions.C, Actions.D
+tables = load_pso_tables("pso_gambler.csv", directory="data")
 
 
-# End with two defections if tournament length is known
-@FinalTransformer((D, D), name_prefix=None)
 class Gambler(LookerUp):
     """
-    A LookerUp class player which will select randomly an action in some cases.
-    It will always defect the last 2 turns.
+    A stochastic version of LookerUp which will select randomly an action in
+    some cases.
     """
 
     name = 'Gambler'
     classifier = {
         'memory_depth': float('inf'),
         'stochastic': True,
-        'makes_use_of': set(['length']),
+        'makes_use_of': set(),
         'long_run_time': False,
         'inspects_source': False,
         'manipulates_source': False,
         'manipulates_state': False
     }
 
-    @init_args
```

Comment:
Actually there's no `__init__` method for gambler at all any more so it's unnecessary.